### ISSUE TITLE
Proposal: Reduce input "daily" restriction a little

### ISSUE THREAD

**Issue by bzah** [Ouranosinc/xclim#882]
* xclim version: master
* Python version: n/a
* Operating System: n/a

### Description

The error `raise ValidationError("time series is not recognized as daily.")` disallow the use of data where the calendar is not homogeneous even if this would not be a problem to compute most indices.
This is particularly true when the data is actually daily but the homogeneity is lost at the hour or minutes scale.

### What I Did

We are doing the beta campaign for icclim v5.
One of our user has met an interesting case with real word data:

Her dataset is _slightly_ inhomogeneous, it goes like:
```
2000-01-01 12:00:00  2000-01-02 11:30:00  2000-01-03 11:30:00  2000-01-04 11:30:00
... 2016-12-29 11:30:00 2016-12-30 11:30:00  2016-12-31 11:30:00
```
So for some reason the first timestamp is at 12:00 but all others are at 11:30.
This is sufficient for the `ValidationError` to be raised.
In this case it's not hard to fix the dataset but:
1. Our user was confused because the error message is wrong, the data is really daily but the issue is at the hour/minute level.
2. I was wondering if xclim could simply ignore this kind of case where values are daily but not homogeneous at a more precise level. 
I was thinking of something like `xclim.config.daily_check_level = "warning"` which would configure xclim to show a warning instead of raising an error. This would be configured by the user.

For the record, I tried with a simple index `xc.atmos.tropical_nights(tasmin=study, freq="MS")` where `study.time` looks like the example above and by commenting the daily check. It _seems_ to work.

### Additional request
3. This is related to https://github.com/Ouranosinc/xclim/issues/867 if I'm not mistaken.
With @pagecp, we would like to allow icclim users to use various kind of input sampling frequencies.
For example if someone has monthly data and want an index at season or year level, he/she should be able to use its data as is.
For now, we have to resample to daily frequency before doing calculation and thus it needs much more computation power than necessary.
In that case, `check_daily` should be replaced by something that check the input data is homogeneous depending on the actual input frequency. 
For example if input is monthly data, only check for year and month consistency but not days, if input precision is minutes by minutes (not sure if it exists) only check for year, month, day, hour and minute consistency and so on.

If needed I would be happy to schedule a call to discuss about this.
Cheers.

**Comment by bzah** [comment]
My point 3. has been mainly solved by #885, thank you @aulemahal. 
For now, Icclim v5 is based on Xclim indicators directly though `atmos` module. 
But I want to migrate to clix-meta.yaml module soon.

Do you intent to implement DailyWeeklyMonthly on Xclim indicators ?
And, would it be possible to apply what you did on anuclim to clix-meta.yaml ?
If so, I'm willing to help.

**Comment by bzah** [comment]
The main topic of this issue is solved by #867

The other topics should probably be discussed in new accurate (and small) gh issues:
- Doc: Add some doc about `xclim.set_option(data_validation="")` in "customizing and controlling xclim" page.
- Feature: Improve error message to reference `xclim.set_option(data_validation="warn")` on raised errors.
- Reopen #649 to accept daily data with incorrect hours ?

If that's ok I'm closing this one a create/reopen the others @aulemahal

**Comment by aulemahal** [comment]
We can do that!

2. Feature : #892 fixed that issue for `check_daily` and `Daily` indicators, but I forgot to edit the error message of `check_freq`, so I need to make that change in #927.
3. We can transfer the discussion back to that issue.
